File: Answer/1189.气球-的最大数量.go
/*
 * @lc app=leetcode.cn id=1189 lang=golang
 *
 * [1189] “气球” 的最大数量
 */

// @lc code=start
func maxNumberOfBalloons(text string) int {
	var cnt [26]int
	for i := 0; i < len(text); i++ {
		cnt[text[i]-'a']++
	}
	b := cnt['b'-'a']
	a := cnt['a'-'a']
	l := cnt['l'-'a'] / 2
	o := cnt['o'-'a'] / 2
	n := cnt['n'-'a']
	if a < b {
		b = a
	}
	if l < b {
		b = l
	}
	if o < b {
		b = o
	}
	if n < b {
		b = n
	}
	return b
}

// @lc code=end

File: Answer/1189.气球-的最大数量.ts
/*
 * @lc app=leetcode.cn id=1189 lang=typescript
 *
 * [1189] “气球” 的最大数量
 */

// @lc code=start
function maxNumberOfBalloons(text: string): number {
    const cnt: number[] = new Array(26).fill(0);
    const aCode = 'a'.charCodeAt(0);
    for (let i = 0; i < text.length; i++) {
        cnt[text.charCodeAt(i) - aCode]++;
    }
    return Math.min(
        Math.min(cnt['b'.charCodeAt(0) - aCode], cnt['a'.charCodeAt(0) - aCode]),
        Math.min(cnt['l'.charCodeAt(0) - aCode] >> 1, cnt['o'.charCodeAt(0) - aCode] >> 1),
        cnt['n'.charCodeAt(0) - aCode]
    );
}
// @lc code=end
